var/spack/repos/builtin/packages/py-pyfr/package.py

```diff
@@ -52,7 +56,48 @@ class PyPyfr(PythonPackage):
 
     # Optional dependencies
     depends_on("py-scipy", when="+scipy", type=("build", "run"))
+    depends_on("metis@5.0:", when="+metis", type=("run"))
     depends_on("scotch@6.0:", when="+scotch", type=("run"))
-    depends_on("cuda@8.0:", when="+cuda", type=("run"))
-    depends_on("rocblas@4.5.0:", when="+hip", type=("run"))
-    depends_on("libxsmm@1.18:+shared blas=0 +large_jit_buffer", when="+libxsmm", type=("run"))
+    depends_on("cuda@8.0: +allow-unsupported-compilers", when="+cuda", type=("run"))
+    depends_on("rocblas@5.2.0:", when="+hip", type=("run"))
+    depends_on("libxsmm@1.18:+shared blas=0", when="+libxsmm", type=("run"))
+
+    # Conflicts for compilers not supporting OpenMP 5.1+ from v1.15.0:
+    conflicts("%gcc@:11", when="@1.15.0: +libxsmm", msg="OpenMP 5.1+ supported compiler required!")
+
+    # Explicitly set library path for dependencies
+    def setup_run_environment(self, env):
+        if "+metis" in self.spec:
+            lib_path = find_libraries(
+                    "libmetis", root=self.spec["metis"].prefix.lib, recursive=False
+                )
+            env.set("PYFR_METIS_LIBRARY_PATH", lib_path[0])
```

Comment:
```suggestion
            env.set("PYFR_METIS_LIBRARY_PATH", self.spec["metis"].libs.directories[0])
```
Packages should always ask other packages where their libraries are installed. Same for all the below packages too.

Reply:
You mean not to assume that they are in a subdir "lib"? Thanks

Reply:
done